File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/domain/Note.java
package at.tugraz.tc.cyfile.domain;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

import java.io.Serializable;
import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Class that represents a Note
 * Created by cobri on 3/21/2018.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public final class Note implements Serializable {

    private String id;

    private String title;

    private String content;

    private Long dateTimeCreated;

    private Long dateTimeModified;

    public Note(String id, String title, String content){
        this.id = id;
        this.title = title;
        this.content = content;
        this.dateTimeCreated = new Date().getTime();
        this.dateTimeModified = dateTimeCreated;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass()) return false;

        Note note = (Note) o;

        return new EqualsBuilder()
                .append(id, note.id)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37)
                .append(id)
                .toHashCode();

    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/domain/Note.kt
package at.tugraz.tc.cyfile.domain

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import java.io.Serializable
import java.util.*


/**
 * Class that represents a Note
 * Created by cobri on 3/21/2018.
 */
class Note(var id: String, var title: String?, var content: String?) : Serializable {

    var dateTimeCreated: Long? = Date().time

    var dateTimeModified: Long?

    override fun equals(other: Any?): Boolean {
        if (this === other) return true

        if (other == null || javaClass != other.javaClass) return false

        val note = other as Note?

        return EqualsBuilder()
                .append(id, note!!.id)
                .isEquals
    }

    override fun hashCode(): Int {
        return HashCodeBuilder(17, 37)
                .append(id)
                .toHashCode()

    }

    init {
        this.dateTimeModified = dateTimeCreated
    }
}
